### ISSUE TITLE
TCPServer Memory Leak

### ISSUE THREAD

**Issue by P3TE** [Unity-Technologies/ROS-TCP-Endpoint#19]
I've noticed a memory leak that occurs whilst running.

I've not verified what the source of the problem, but I think it has to do with the following code:
RosTCPServer.py, line 51:
`

        threads = []
        while True:
            tcp_server.listen(self.connections)

            (conn, (ip, port)) = tcp_server.accept()
            new_thread = ClientThread(conn, self, ip, port)
            new_thread.start()
            threads.append(new_thread)

        for t in threads:
            t.join()`

I believe the problem is that each published message opens a new TCP connection, starting a new thread which is appended to a list of threads, but they are never being removed until the program has stopped.
Also I'm not sure how the thread joins are ever called when they are outside of a 'while True:'

**Comment by at669** [comment]
Hey @P3TE! We'll take a look into this. We realized the `for t in threads:` block is unreachable--this has been addressed in our `dev` branch.

**Comment by P3TE** [comment]
So I've created a fix for this on a fork of this repo.
In fact I've taken a slightly different approach for the TCP connection, I've implemented a persistent connection for each published topic.
Note that so far this only implemented on my fork for publishers and I've not implemented services or subscribers.
In any case, feel free to look at the changes I've made on these forks and branches for inspiration on how to fix the memory leak.
Endpoint:
https://github.com/P3TE/ROS-TCP-Endpoint/tree/persistent-tcp-connections
TCP Connector:
https://github.com/P3TE/ROS-TCP-Connector/tree/persistent-tcp
